plugins/op-z/export.js

```diff
@@ -0,0 +1,130 @@
+const {
+  CoarseChannel,
+  FineChannel,
+  NullChannel,
+  SwitchingChannel
+} = require(`../../lib/model.js`);
+
+module.exports.name = `OP-Z`;
+module.exports.version = `0.1.0`;
+
+const MAX_KNOBS = 8;
+const MAX_OPZ_FIXTURES = 16;
+
+/**
+ * @param {array.<Fixture>} fixtures An array of Fixture objects.
+ * @param {object} options Global options, including:
+ * @param {string} options.baseDir Absolute path to OFL's root directory.
+ * @param {Date|null} options.date The current time.
+ * @returns {Promise.<array.<object>, Error>} The generated files.
+*/
+module.exports.export = function exportOpZ(fixtures, options) {
+  const exportJson = {
+    profiles: [],
+    config: []
+  };
+
+  const usedKnobs = {};
+
+  fixtures.forEach(fixture => {
+    const fixtureKey = `${fixture.manufacturer.key}/${fixture.key}`;
+
+    fixture.modes.forEach((mode, modeIndex) => {
+      const modeName = `${fixtureKey}/${mode.shortName}`;
+
+      // add profile
+      exportJson.profiles.push({
+        name: modeName,
+        channels: mode.channels.map(channel => getOpZChannelType(channel, fixtureKey))
+      });
+
+      // add config
+      if (modeIndex === 0 && exportJson.config.length < MAX_OPZ_FIXTURES) {
+        exportJson.config.push({
+          fixture: exportJson.config.length + 1,
+          profile: modeName
+        });
+      }
+    });
+  });
+
+
+  return Promise.resolve([{
+    name: `dmx.json`,
+    content: JSON.stringify(exportJson, null, 2),
+    mimetype: `application/json`,
+    fixtures
+  }]);
+
+
+
+  /**
+   * @param {AbstractChannel} channel The OFL channel object.
+   * @param {string} fixtureKey The OFL fixture key.
+   * @returns {string} The OP-Z channel type.
+   */
+  function getOpZChannelType(channel, fixtureKey) {
+    if (channel instanceof SwitchingChannel) {
+      channel = channel.defaultChannel;
+    }
+
+    if (channel instanceof NullChannel || channel instanceof FineChannel) {
+      return `off`;
+    }
+
```

Comment:
Constant channels should have higher priority than opZChannelTypes and therefore should return the default value already at this point here.

Reply:
Fixed in 6f49a66.